src/components/navbar/Navbar.jsx

```diff
@@ -20,21 +24,40 @@ const navbarLinks = [
     name: "FAQs",
   },
 ];
-
 const Navbar = () => {
+  const [isNavOpen, setIsNavOpen] = useState(false);
+  const [width,setWidth]=useState(window.innerWidth);
+  useEffect(() => {
+    const handleResize = () => {
+      setWidth(window.innerWidth);
+    };
+
+    // Add a resize event listener
+    window.addEventListener('resize', handleResize);
+
+    // Clean up the event listener when the component unmounts
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  }, []);
+  const toggleNav = () => {
+    setIsNavOpen(!isNavOpen);
```

Comment:
```suggestion
    setIsNavOpen(isNavOpen => !isNavOpen);
```

Refer : https://legacy.reactjs.org/docs/hooks-reference.html#functional-updates